erdpy/validators/validators.py

```diff
@@ -18,18 +26,35 @@ def estimate_system_sc_call(args, base_cost, factor=1):
     return gas_limit
 
 
+def _read_json_file(file_path):
+    val_file = path.expanduser(file_path)
+    guards.is_file(val_file)
+    with open(file_path,  "r") as json_file:
+        try:
+            data = json.load(json_file)
+        except Exception:
+            raise Exception("cannot read validators data")
+        return data
+
+
 def parse_args_for_stake(args):
-    num_of_nodes = int(args.number_of_nodes)
-    keys = args.nodes_public_keys.split(',')
-    if num_of_nodes != len(keys):
-        raise errors.BadUserInput("check number of nodes")
-        return
+    validators_data_file = args.validators_data_file
+    validators_data = _read_json_file(validators_data_file)
 
     reward_address = args.reward_address
 
+    if args.pem:
+        account = Account(pem_file=args.pem)
+    elif args.keyfile and args.passfile:
+        account = Account(key_file=args.keyfile, pass_file=args.passfile)
+
+    num_of_nodes = len(validators_data["validators"])
     stake_data = 'stake@' + binascii.hexlify(num_of_nodes.to_bytes(1, byteorder="little")).decode()
-    for key in keys:
-        stake_data += '@' + key + '@' + convert_to_hex('genesis')
+    for validator in validators_data["validators"]:
+        # get validator
+        seed, bls_key = parse_validator_pem(validator["pemFilePath"])
```

Comment:
don't we need any validation, at least keys size?

Reply:
i think not because in that file CLI user have to provided a path to pem file. And in pem file length of key should be fine.